e2e/portfolio.spec.ts

```diff
@@ -27,6 +28,13 @@ test.describe('Add Portfolio Item', () => {
     await page
       .getByRole('button', { name: 'Add a new portfolio Item' })
       .click();
+
+    // Will check if the portfolio button is hydrated correctly with different intervals.
+    await expect(async () => {
+      await expect(
+        page.getByRole('button', { name: 'Add a new portfolio Item' })
+      ).toBeDisabled();
+    }).toPass();
```

Comment:
```suggestion

    // Will check if the portfolio button is hydrated correctly with different intervals.
    await expect(async () => {
      const addPortfolioItemButton = page.getByRole('button', {
        name: 'Add a new portfolio Item'
      });
      await addPortfolioItemButton.click();

      await expect(addPortfolioItemButton).toBeDisabled({ timeout: 1 });
    }).toPass();
```
So, the thing is, there's no point re-checking if the button is disabled without clicking it. The problem is that if playwright clicks too early, the button click doesn't do anything, so waiting for even longer (i.e. what `toPass` does) just makes the test take longer to fail.

The way to think about `toPass` is that it lets you retry something unreliable (clicking) until it finally succeeds (button is disabled). You have to do both in the callback for that to work.

Reply:
I must not have been thinking straight 🙃